hooks/usePaginatedEvents.ts

```diff
@@ -13,10 +13,10 @@ export function usePaginatedEvents(
   fetchNext: () => void
 } {
   const $hasPrevious = useMemo(
-    () => events?.metadata.has_previous ?? false,
+    () => events?.metadata?.has_previous ?? false,
     [events]
   )
-  const $hasNext = useMemo(() => events?.metadata.has_next ?? false, [events])
+  const $hasNext = useMemo(() => events?.metadata?.has_next ?? false, [events])
```

Comment:
Let's not use this operator unless TypeScript requires us to. If there's code in the API that can return metadata as undefined, I'd like to take a look at that, and we can update the TS types for `API.ListEventsResponse`. Otherwise, I'd definitely consider it an error in the frontend if it's somehow undefined in practice, and we should be surfacing the error rather than coalescing it to false.

Reply:
@dguenther I'm a little confused by this comment -- it seems like we were already using all the operators, I just added one level of resolving potentially undefined values here. I understand if your argument is "let's have all impossible states fail", but from my perspective we were already covering a single impossible case, now we're coalescing a definite value regardless of anything. Am I off here?

Reply:
When I mentioned "unless TypeScript requires us to," I meant let's only use it if the value has the possibility to be `null` or `undefined`. In this case, `events` has the possibility to be undefined, but if `events` is defined, `events.metadata` will never be undefined (based on the TypeScript types).

Oh, maybe it wasn't clear that the `?.` operator is short-circuiting? The rest of the expression won't be evaluated if a `?.` operator is used on a null/undefined value.

Reply:
Ahhhhhh. I did not know that it was a short-circuiting operation. That makes tons of sense.